xline/src/storage/alarm_store.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicI32, Ordering},
+        Arc,
+    },
+};
+
+use curp::members::ServerId;
+use parking_lot::RwLock;
+use prost::Message;
+use xlineapi::{
+    command::{CommandResponse, SyncResponse},
+    execute_error::ExecuteError,
+    AlarmAction, AlarmMember, AlarmResponse, AlarmType, RequestWithToken, RequestWrapper,
+    ResponseWrapper,
+};
+
+use super::{db::WriteOp, storage_api::StorageApi};
+use crate::header_gen::HeaderGenerator;
+
+/// Alarm table name
+pub(crate) const ALARM_TABLE: &str = "alarm";
+
+/// Alarm store
+#[derive(Debug)]
+pub(crate) struct AlarmStore<DB>
+where
+    DB: StorageApi,
+{
+    /// Header generator
+    header_gen: Arc<HeaderGenerator>,
+    /// Persistent storage
+    db: Arc<DB>,
+    /// Alarm types
+    types: RwLock<HashMap<AlarmType, HashMap<ServerId, AlarmMember>>>,
```

Comment:
This could be replaced by `DashMap`.

Reply:
The behaviors of `DashMap` and `RwLock<HashMap>`are not exactly the same, using `DashMap` here will cause the alarm to not work properly